src/test/javascript/spec/component/exercises/shared/result.component.spec.ts

```diff
@@ -52,11 +70,30 @@ describe('ResultComponent', () => {
         expect(component).not.toBeNull();
     });
 
-    it('should set results', () => {
-        const submission1 = { id: 1 } as Submission;
-        const result1 = { id: 1, submission: submission1, score: 1 } as Result;
-        const result2 = { id: 2 } as Result;
-        const participation1 = cloneDeep(participation);
+    it('should set results for programming exercise', () => {
+        const submission1: Submission = { id: 1 };
+        const result1: Result = { id: 1, submission: submission1, score: 1 };
+        const result2: Result = { id: 2 };
+        const participation1 = cloneDeep(programmingParticipation);
+        participation1.results = [result1, result2];
+        component.participation = participation1;
+
+        fixture.detectChanges();
+
+        expect(component.result).toEqual(result1);
+        expect(component.result!.participation).toEqual(participation1);
+        expect(component.submission).toEqual(submission1);
+        expect(component.textColorClass).toEqual('text-secondary');
```

Comment:
```suggestion
        expect(component.textColorClass).toBe('text-secondary');
```
Primitive types should be compared strictly.